pipelines/build/common/build_base_file.groovy

```diff
@@ -877,12 +879,19 @@ class Builder implements Serializable {
                                         } catch (FlowInterruptedException e) {
                                             throw new Exception("[ERROR] Archive artifact timeout (${pipelineTimeouts.ARCHIVE_ARTIFACTS_TIMEOUT} HOURS) for ${downstreamJobName}has been reached. Exiting...")
                                         }
+
+                                        copyArtifactSuccess = true
                                     }
+                            }
+                            context.println '[NODE SHIFT] OUT OF CONTROLLER NODE!'
+
+                            if ((downstreamJob.getResult() != 'SUCCESS' || !copyArtifactSuccess) && propagateFailures) {
+                                context.error("Propagating downstream job result: ${downstreamJobName}, Result: "+downstreamJob.getResult()+" CopyArtifactsSuccess: "+copyArtifactSuccess)
+                                if (copyArtifactSuccess && downstreamJob.getResult() == 'UNSTABLE' && (currentBuild.result == 'SUCCESS' || currentBuild.result == 'UNSTABLE' )) {
```

Comment:
As for this logic: if pipeline has 3 downstream build jobs.
the 1st one finishes with **non-success** or **with success but failed to copy artifacts**, it immediately sets the pipeline status to **failure** or **unstable** , in this case, the whole pipeline will be exit i.e the other two downstream jobs get killed even they are still on-going?
Do i understand this correctly?

Reply:
It doesn't exit, it just sets the pipeline result to Failure. The "jobs parallel" has to completely finish before the pipeline will exit.

Reply:
So I confirmed, by aborting one, that the other builds continued, nothing was killed...